src/core/Quote.ml

```diff
@@ -267,6 +271,39 @@ let rec quote_con (tp : D.tp) con =
     Format.eprintf "bad: %a / %a@." D.pp_tp tp D.pp_con con;
     throw @@ QuotationError (Error.IllTypedQuotationProblem (tp, con))
 
+and quote_fields (sign : D.sign) (fields : (string * D.con) list) : (string * S.t) list m =
+  match sign, fields with
+  | D.Field (lbl, tp, sign_clo), (flbl, fcon) :: fields ->
+     (* FIXME: Computation??? *)
+     let* sign = lift_cmp @@ inst_sign_clo sign_clo fcon in
+     let* tfield = quote_con tp fcon in
+     let+ tfields = quote_fields sign fields in
+     (lbl, tfield) :: tfields
+  | D.Empty, [] -> ret []
+  | _, _ -> failwith "FIXME: Better Error handling on sig/field mismatch in 'quote_fields'"
+
+(* Given a signature:
```

Comment:
So one high-level question I have is that it seems like the association of cells in the signature is here in the `snoc telescope` style, and elsewhere it is in the `cons telescope` style. Personally I am not sure which is best for our application, but I wanted to hear your thoughts.

Reply:
I don't like this either haha. The more code I write, the more snoc is starting to seem like the natural fit, with the exception that all recursion wants to start from the LHS of the telescope (I never know how to name the folds over snoc lists haha)

The pattern that seems to be arising the most often is that we perform some sort of map over the fields where we need to carry along an accumulator that ought to be a snoc telescope. Once all the details are hammered out I'm going to factor a bunch of the bespoke recursion into a set of combinators and then just use those, should make the muddy waters a bit clearer.

Reply:
I'm reversing my position here! I messed around with using `bwd` for our telescopes, but the thing that makes this not as natural as I thought is the interplay with `D.sign`:
```
  and sign =
    | Field of string * tp * S.sign clo
    | Empty
```
Going between these two structures when `S.sign` is a snoc list doesn't feel quite right, and is super obvious when we have cons lists, so it seems that those really are the right choice.